docs/core/allocation.qmd

```diff
@@ -0,0 +1,178 @@
+
+## Allocation problem input
+
+### The subnetwork
+
+The allocation problem is solved per subgraph, where a subgraph is given by a subset $S \subset V$ of node ids. Different subgraphs are disjoint from eachother.
+
+### Source flows
+
+Sources are indicated a set of edges in the subnetwork
+$$
+E_S^\text{source} \subset \left(S \times S\right) \cap E.
+$$
+That is, if $(i,j) \in E_S^\text{source}$, then $Q_{ij}$ is treated as a source flow in the allocation problem.
+
+### User demands
+
+The subnetwork contains a subset of user nodes $U_S \subset S$, who all have time varying demands over various priorities $p$:
+$$
+    d^p_i(t), \quad i \in U_S, p = 1,2,\ldots, p_{\max}.
+$$
+
+:::{.callout-note}
+On this page we assume that the priorities are given by all integers from $1$ to some $p_{\max} \in \mathbb{N}$. However, in the Ribasim input this is not a requirement; some of these in between priority values can be missing, only the ordering of the given priorities is taken into account.
+:::
+
+### Vertical fluxes and local storage
+
+Apart from the source flows denoted by edges, there are other sources of water in the subnetwork, associated with the basins in the subnetwork $B_S = B \cap S$. Firstly there is the sum of the horizontal fluxes (precipitation, evaporation, infiltration and drainage) for each basin
+$$
+    \phi_i(t), \quad \forall i \in B_S.
+$$
+
+Secondly, there is the available water in each basin above the minimum level $l_{\min,i}$
+$$
+    u_i(t)-S(l_{\min,i}), \quad \forall i \in B_S.
+$$
+Note that this value can be negative, which we interpret as a demand from the basin.
+
+### Flow magnitude and direction constraints
+
+Nodes in the Ribasim model that have a `max_flow_rate`, i.e. pumps and outlets, put a constraint on the flow through that node. Some nodes only allow flow in one direction, like pumps, outlets and tabulated rating curves.
+
+### Fractional flows and user return flows
+
+Both fractional flow nodes and user nodes dictate proportional relationships between flows over edges in the subnetwork. Users have a return factor $r_i, i \in U_S$.
+
+## The allocation optimization problem
+
+### The allocation graph
+
+A new graph is created from the subnetwork, which we call the allocation graph. To indicate the difference between subnetwork data and allocation graph data, the allocation graph data is denoted with a hat. The allocation graph consists of:
+
+- Nodes $\hat{V_S}$, where each basin, source and user in the subnetwork get a node in the allocation graph. Also nodes that have fractional flow outneighbors get a node in the allocation graph. The implementation makes heavy use of the node id mapping $m_S : i \mapsto \hat{i}$ to translate from subnetwork node Ds to allocation graph node IDs. Unless specified otherwise, we assume this relationship between index symbols that appear both with and without a hat.
```

Comment:
```suggestion
- Nodes $\hat{V_S}$, where each basin, source and user in the subnetwork get a node in the allocation graph. Also nodes that have fractional flow outneighbors get a node in the allocation graph. The implementation makes heavy use of the node id mapping $m_S : i \mapsto \hat{i}$ to translate from subnetwork node IDs to allocation graph node IDs. Unless specified otherwise, we assume this relationship between index symbols that appear both with and without a hat.
```